test/evergreen.yml

```diff
@@ -216,16 +216,18 @@ functions:
           set -o errexit
           set -o verbose
 
-          # Check whether specific branches have been provided to the function
-          if [ -z ${branches} ]; then
-            wt_branches=${branch_name}
+          # Check whether specific branches have been provided to the function through the task
+          # variable defined in doc-update. If none have been specified, the built-in Evergreen
+          # expansion ${branch_name} is used.
+          if [ -z ${doc_update_branches} ]; then
+            branches=${branch_name}
           else
-            wt_branches=${branches}
+            branches=${doc_update_branches}
```

Comment:
We should also use the `${doc_update_branches}` variable in "update wiredtiger docs" - the branch names are still hardcoded there.

Reply:
I was not sure at first since `update wiredtiger docs` is used in a very specific context while `compile wiredtiger docs` is used in different ones but I agree, it is better than having hardcoded values. I decided to go for the expansion variable, let me know what you think, see 7e2e1867925c3126f62ee929ce70d6c02a7a1bb8.

Reply:
I don't think it makes sense to do one without the other? Or rather, there should be the same set of branches for both.

Oh wait, I don't think the `doc_compile` task will work as expected. You are setting the `doc_update_branches` variable in the build variant but the `doc_compile` task is set to run outside the variant, in PR testing.

Reply:
> I don't think it makes sense to do one without the other? Or rather, there should be the same set of branches for both.

To make sure they have the same set of branches for both, we can either have an expansion variable or use the same input for both functions.

 > Oh wait, I don't think the doc_compile task will work as expected. You are setting the doc_update_branches variable in the build variant but the doc_compile task is set to run outside the variant, in PR testing.
 
Yes, in PR testing `doc_compile` will run in a different variant where `doc_update_branches` is not defined, hence it will use `${branch_name}` as expected.

Reply:
Ak okay, got it!